lua/nvim-treesitter/textobjects/move.lua

```diff
@@ -70,16 +74,157 @@ local function move(query_strings, forward, start, winid)
 end
 
 M.goto_next_start = function(query_strings)
-  move(query_strings, "forward", "start")
+  local args = { query_strings, "forward", "start" }
+  move(unpack(args))
+  M.last_move = { func = move, args = args }
 end
 M.goto_next_end = function(query_strings)
-  move(query_strings, "forward", not "start")
+  local args = { query_strings, "forward", not "start" }
+  move(unpack(args))
+  M.last_move = { func = move, args = args }
 end
 M.goto_previous_start = function(query_strings)
-  move(query_strings, not "forward", "start")
+  local args = { query_strings, not "forward", "start" }
+  move(unpack(args))
+  M.last_move = { func = move, args = args }
 end
 M.goto_previous_end = function(query_strings)
-  move(query_strings, not "forward", not "start")
+  local args = { query_strings, not "forward", not "start" }
+  move(unpack(args))
+  M.last_move = { func = move, args = args }
+end
+
+-- implements naive f, F, t, T with repeat support
+local function builtin_find(forward, inclusive, char, repeating)
+  -- forward = true -> f, t
+  -- inclusive = false -> t, T
+  -- if repeating with till (t or T, inclusive = false) then search from the next character
+  -- returns nil if cancelled or char
+  char = char or vim.fn.nr2char(vim.fn.getchar())
+  repeating = repeating or false
+
+  if char == vim.fn.nr2char(27) then
+    -- escape
+    return nil
+  end
+
+  local line = vim.api.nvim_get_current_line()
+  local cursor = vim.api.nvim_win_get_cursor(0)
```

Comment:
our move functions can use `winid`. Wouldn't it be useful to have builtin_find support it as well?

Reply:
Done!